src/FritzBox/Converter.php

```diff
@@ -215,57 +225,56 @@ private function getPhoneNumbers(stdClass $card): array
             });
         }
 
-        return $res;
+        return $phoneNumbers;
     }
 
     /**
      * Return an array of prequalified email adresses. There is no limitation
      * for the amount of email adresses in FRITZ!Box phonebook contacts.
      *
-     * @param stdClass $card
+     * @param mixed $card
      * @return array
      */
-    private function getEmailAdresses(stdClass $card): array
+    private function getEmailAdresses($card): array
     {
-        if (!isset($card->email)) {
+        if (!isset($card->EMAIL)) {
             return [];
         }
 
         $mailAdresses = [];
         $emailTypes = $this->config['emailTypes'] ?? [];
 
-        foreach ($card->email as $emailType => $addresses) {
-            foreach ($addresses as $addr) {
-                $addAddress = [
-                    'id' => count($mailAdresses),
-                    'email' => $addr,
-                ];
-
-                foreach ($emailTypes as $type => $value) {
-                    if (stripos($emailType, $type) !== false) {
-                        $addAddress['classifier'] = $value;
-                        break;
-                    }
+        foreach ($card->EMAIL as $key => $address) {
+            $addAddress = [
+                'id'    => count($mailAdresses),
+                'email' => (string)$address,
+            ];
+
+            $mailTypes = isset($card->EMAIL[$key]->parameters['TYPE']) ? strtoupper($card->EMAIL[$key]->parameters['TYPE']) : '';
```

Comment:
Use ?: to simplify

Reply:
dito
this way?:
$mailTypes = strtouppper(isset($card->EMAIL[$key]->parameters['TYPE'])) ?: '';

Reply:
Almost- last brackts need to move before ;

Reply:
dito

Reply:
see below: $telTypes = ...